scarb/tests/metadata.rs

```diff
@@ -259,6 +272,78 @@ fn dev_dependencies() {
     );
 }
 
+#[test]
+#[ignore = "not implemented yet"]
+fn dev_deps_are_not_propagated() {
+    // TODO(maciektr): Make sure dev-deps are not propagated.
+    let t = assert_fs::TempDir::new().unwrap();
+
+    let dep1 = t.child("dep1");
+    ProjectBuilder::start().name("dep1").build(&dep1);
+
+    let dep2 = t.child("dep2");
+    ProjectBuilder::start()
+        .name("dep2")
+        .dev_dep("dep1", &dep1)
+        .build(&dep2);
+
+    let pkg = t.child("pkg");
+    ProjectBuilder::start()
+        .name("x")
+        .dep("dep2", &dep2)
+        .build(&pkg);
+
+    let metadata = Scarb::quick_snapbox()
+        .arg("--json")
+        .arg("metadata")
+        .arg("--format-version")
+        .arg("1")
+        .current_dir(&pkg)
+        .stdout_json::<Metadata>();
+
+    assert_eq!(
+        packages_and_deps(metadata.clone()),
+        BTreeMap::from_iter([
+            ("core".to_string(), vec![]),
+            ("test_plugin".to_string(), vec![]),
+            (
+                "x".to_string(),
+                vec![
+                    "core".to_string(),
+                    "dep2".to_string(),
```

Comment:
Can you please add a comment here stating that without exclusion, there would be `dep1` here?